### DISCUSSION TITLE
New GUI ethLoader

### DISCUSSION THREAD

**Discussion by hu-yue** [robotology/community#151] [Q&A] [answered]
Hi, 

I wanted to update the robot, and as I've understood from issues #134 and #135, there is a new gui for the `ethLoader`, and all the updated instructions are referred to the new gui.

However, even if I updated everything, I still see the old gui; therefore, is there any option or so that I had to activate to see the new gui?

(I updated the firmware anyway with the old one, but for future and also students it would be better to have the one that matches with the instructions).

Thanks!

**Comment by marcoaccame** [comment, marked as answer]
Hi @hu-yue, in master there is not really a new GUI for ethLoader. 

Some months ago, I just did a minor change to the layout: I reorganized the column to show some more information and I have renamed some buttons. 

Following this change, new instructions were also required. The instructions refer to the new layout because one should typically perform an update of everything, as described in https://github.com/robotology/QA/issues/135.

So, the ethLoader GUI is not new, it just shows some more information. However, to have the new information displayed one also need to update the FW of the eLoader and eUpdater to versions equal or higher than 2.20. 

Should one update also the FW of eLoader and eUpdater? I did not mention it because it is not necessary yet. See issue https://github.com/robotology/QA/issues/134 for reasons.

Here are some screenshot so that you can better understand the changes of layout and what happens if you update eLoader / eUpdater or not. 

![image](https://cloud.githubusercontent.com/assets/7148284/19637984/046a9c8c-99d2-11e6-9adf-2188b3e5c372.png)

**Figure 1** - Previous GUI for ethLoader.

![image](https://cloud.githubusercontent.com/assets/7148284/19638483/7c612be6-99d4-11e6-9452-1f893a9dc58f.png)

**Figure 2** - Current GUI for ethLoader.  It has extra columns about type of board, running process, date of FW and user-defined info of the board. Board 10.0.1.1 does not show the new information because it has old versions of eLoader and eUpdater, whereas the other boards do because they have  eLoader / eUpdater with version  >= 2.20.

**Comment by hu-yue** [comment]
@marcoaccame thanks for the clarification!
We will try to update the eLoader and eUpdater later on, I will close this issue for the time being.

**Comment by marcoaccame** [comment]
Hi @hu-yue ,

when you update eLoader, **remember** that instructions tells: 
`In doing this operation, be careful and update only one board at a time because with some upgrades the  board may change its address into 10.0.1.99`.
Well, this update does force a reset of the address, hence ... do it one board at a time otherwise all boards will have the same IP address.
Also, if you change the IP address, you need to restart the board to see the new address in the GUI.
